walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,126 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public static final int DAY_START = TimeRange.START_OF_DAY;
+    public static final int DAY_END = TimeRange.END_OF_DAY;
+    public static final TimeRange WHOLE_DAY = TimeRange.WHOLE_DAY;
+
+    /**
+      * Return a collection of TimeRange that attendees can meet for a requested 
+      * meeting. If there is not enough room for optional attendees, find TimeRanges 
+      * that just work for required attendees. If there are no required attendees, 
+      * find the TimeRangs for just the optional attendees.
+      */
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        long meetingDuration = request.getDuration();
+        // Required attendees for the request meeting.
```

Comment:
I don't think this comment adds much beyond what the `requiredAttendees` name already conveys. Same is true for the comment for `allAttendees`. I also think 'the requested meeting' sounds better than 'the request meeting'.

Reply:
The redundant comments are removed! :)